src/RA_AchievementSet.cpp

```diff
@@ -17,36 +14,44 @@
 #include "services\ILeaderboardManager.hh"
 #include "services\ServiceLocator.hh"
 
+#include <array>
+#include <fstream>
+#include <memory>
+
 AchievementSet* g_pCoreAchievements = nullptr;
 AchievementSet* g_pUnofficialAchievements = nullptr;
 AchievementSet* g_pLocalAchievements = nullptr;
 
-AchievementSet** ACH_SETS[] ={ &g_pCoreAchievements, &g_pUnofficialAchievements, &g_pLocalAchievements };
-static_assert(SIZEOF_ARRAY(ACH_SETS) == NumAchievementSetTypes, "Must match!");
+inline constexpr std::array<AchievementSet**, 3> ACH_SETS
+{
+    &g_pCoreAchievements,
+    &g_pUnofficialAchievements,
+    &g_pLocalAchievements
+};
 
-AchievementSetType g_nActiveAchievementSet = Core;
+AchievementSet::Type g_nActiveAchievementSet = AchievementSet::Type::Core;
 AchievementSet* g_pActiveAchievements = g_pCoreAchievements;
 
-
-void RASetAchievementCollection(AchievementSetType Type)
+_Use_decl_annotations_
+void RASetAchievementCollection(AchievementSet::Type Type) noexcept
 {
     g_nActiveAchievementSet = Type;
-    g_pActiveAchievements = *ACH_SETS[Type];
+    g_pActiveAchievements = *ACH_SETS.at(ra::etoi(Type));
 }
 
-std::wstring AchievementSet::GetAchievementSetFilename(ra::GameID nGameID)
+_Use_decl_annotations_
+std::wstring AchievementSet::GetAchievementSetFilename(ra::GameID nGameID) noexcept
 {
-    switch (m_nSetType)
+    std::wstring str;
     {
-        case Core:
-            return g_sHomeDir + RA_DIR_DATA + std::to_wstring(nGameID) + L".txt";
-        case Unofficial:
-            return g_sHomeDir + RA_DIR_DATA + std::to_wstring(nGameID) + L".txt";	// Same as Core
-        case Local:
-            return g_sHomeDir + RA_DIR_DATA + std::to_wstring(nGameID) + L"-User.txt";
-        default:
-            return L"";
+        std::wostringstream oss;
+        oss << g_sHomeDir << RA_DIR_DATA << nGameID;
+        if (m_nSetType == Type::Local)
+            oss << L"-User";
+        oss << L".txt";
+        str = oss.str();
     }
+    return str;
```

Comment:
Gives it a stronger guarantee that an exception will not be thrown, got rid of the switch since it's already handled in the scope.

The `str = oss.str()` is a move assign and they aren't allowed to throw.